ofrak_core/test_ofrak/components/test_patch_maker_component.py

```diff
@@ -47,6 +48,14 @@
 ARM32_PROGRAM_PATH = str(Path(__file__).parent / "assets" / "simple_arm_gcc.o.elf")
 
 
+@pytest.fixture(params=[EphemeralResourceContextFactory, ClientResourceContextFactory])
+async def ofrak_context(request, ofrak):
```

Comment:
I suggest adding a comment here explaining why the patch maker component tests get a special OFRAK context. I'm inferring that it's just because the components are somewhat complicated so they probably exercise the different resource context types.

Also, is it possible to reference the "super" fixture of the same name, so it's more clear it is just wrapping it rather than doing anything significantly different?

```
async def ofrak_context(request, ofrak_context):
    ofrak_context._resource_context_factory = request.param()
    return ofrak_context
```

Does that work?

Reply:
@whyitfor told me to only override the fixture here because the original bug was triggered in these tests. It is scoped to a file because when this fixture is used, it will cause all tests that use it to be run twice. Instead of having all tests run with both factories, we only run these tests with both factories. 

I'll try the wrapping you suggest.